### ISSUE TITLE
`enum_variant_names` lint triggers too easily

### ISSUE THREAD

**Issue by sdroege** [rust-lang/rust-clippy#8270]
### Summary

See below enum for an example. It only complains about one of the variants, and should probably complain about neither because how else would you call these? :)

### Lint Name

enum_variant_names

### Reproducer

I tried this code:

```rust
#[derive(Debug, Clone)]
enum Message {
    Buffer(gst::Buffer),
    Event(gst::Event),
    Message(gst::Message),
    ServerError(String),
}
```

I saw this happen:

```
warning: variant name ends with the enum's name
  --> net/reqwest/tests/reqwesthttpsrc.rs:42:5
   |
42 |     Message(gst::Message),
   |     ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::enum_variant_names)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names
```

I expected to see this happen: Nothing

### Version

```text
rustc 1.58.0 (02072b482 2022-01-11)
binary: rustc
commit-hash: 02072b482a8b5357f7fb5e5637444ae30e423c40
commit-date: 2022-01-11
host: x86_64-unknown-linux-gnu
release: 1.58.0
LLVM version: 13.0.0
```

**Comment by sdroege** [comment]
Also still happens with current nightly.

**Comment by dswij** [comment]
Unless I'm missing something, it only complains about one of the variant because there's only one that starts (or ends) with `Message`, and the help message says ends rather than starts because of the order it checks.

I think that this lint is behaving correctly [as intended](https://rust-lang.github.io/rust-clippy/master/)?

**Comment by sdroege** [comment]
Ah it's complaining about the name of the enum, not about the name of the type the tuple variant contains.

Yes then it behaves correctly and maybe here the suggestion should be to rename the enum to something more descriptive, but I'm fine with just ignoring the lint here as any other name I can come up with is less useful :)
